### DISCUSSION TITLE
Evolution of CL since 2019, as seen by OAK

### DISCUSSION THREAD

**Discussion by gouttegd** [obophenotype/cell-ontology#2416] [General]
As part of the work towards the next “CL update” paper, here are some OAK-generated “diffs“ between the last release of CL (2024-05-15) and some older versions.

All diffs are generated on the `base` product variant.

## From 2019-04-05 release to 2024-05-15

The 2019-04-05 was the first ODK-managed release. As such, it’s the first release a `cl-base` product.

(I wanted to post the Markdown-formatted version of the diff directly “inlined” in this post, so that it would be easily viewable, but it’s way too long for GitHub’s post limits. So I attach it to the post instead; it’s much less useful that way — I believe the CSV version is better for download and viewing locally – but that’s all I can do.)

Markdown version: [diff-20190405-20240515.md](https://github.com/user-attachments/files/16053278/diff-20190405-20240515.md)

CSV version of the diff (with changes represented as KGCL changes) for easy manipulation: [diff-20190405-20240515.csv](https://github.com/user-attachments/files/16052362/diff-20190405-20240515.csv)

**Comment by gouttegd** [comment]
## From 2021-04-22 to 2024-05-15

Markdown version: [diff-20210422-20240515.md](https://github.com/user-attachments/files/16053298/diff-20210422-20240515.md)

CSV version: [diff-20210422-20240515.csv](https://github.com/user-attachments/files/16053301/diff-20210422-20240515.csv)

**Comment by gouttegd** [comment]
## From 2023-05-22 to 2024-05-15

Markdown version: [diff-20230522-20240515.md](https://github.com/user-attachments/files/16053316/diff-20230522-20240515.md)

CSV version: [diff-20230522-20240515.csv](https://github.com/user-attachments/files/16053317/diff-20230522-20240515.csv)

**Comment by dosumis** [comment]
I see this are too large for tickets, but would be good to post somewhere on GitHub to take advantage of folded reports for easy browsing.

**Reply by gouttegd** [reply]
I am not very keen on “polluting” the main CL repository with huge `.md` files, but how about creating a dedicated repository (probably in the `obophenotype` organisation) for that? It could be used for hosting similar files, not only for CL, but also for others obophenotype ontologies (e.g. Uberon) when we need them.

**Comment by dosumis** [comment]
The new relationship numbers look suspiciously large - 4948.  Are we sure we're comparing like-for-like?  If using base, this could be inflated by the move to adding inferences to base.

**Reply by gouttegd** [reply]
All comparisons are made against the `-base` product for every release — **using whatever definition of -base was used by the ODK at the time of release**.

I’ve made new diffs against the `-non-classified` versions after forcefully removing non-CL terms.

* From 2020-05-20 to 2024-05-15
  * [diff-oak-20200520-20240515.csv](https://github.com/user-attachments/files/16197572/diff-oak-20200520-20240515.csv)
  * [diff-oak-20200520-20240515.md](https://github.com/user-attachments/files/16197573/diff-oak-20200520-20240515.md)

* From 2021-04-22 to 2024-05-15
  * [diff-oak-20210422-20240515.csv](https://github.com/user-attachments/files/16197590/diff-oak-20210422-20240515.csv)
  * [diff-oak-20210422-20240515.md](https://github.com/user-attachments/files/16197591/diff-oak-20210422-20240515.md)

* From 2023-05-22 to 2024-05-15
  * [diff-oak-20230522-20240515.csv](https://github.com/user-attachments/files/16197577/diff-oak-20230522-20240515.csv)
  * [diff-oak-20230522-20240515.md](https://github.com/user-attachments/files/16197578/diff-oak-20230522-20240515.md)

**Reply by gouttegd** [reply]
Can’t do a comparison with the 2019 releases as the `non-classified` artefacts were not published back then.
